src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -508,8 +508,13 @@ static UCS_F_ALWAYS_INLINE void uct_ib_mlx5_bf_copy_bb(void * restrict dst,
     UCS_WORD_COPY(__m128i, dst, __m128i, src, MLX5_SEND_WQE_BB);
 #elif defined(__ARM_NEON)
     vst4q_u64(dst, vld4q_u64(src));
-#else /* NO SIMD support */
-    UCS_WORD_COPY(uint64_t, dst, uint64_t, src, MLX5_SEND_WQE_BB);
+#else
+    typedef struct {
+        uint8_t data[MLX5_SEND_WQE_BB];
+    } UCS_S_PACKED uct_ib_mlx5_send_wqe_bb_t;
+
+    /* Prevent the compiler to replace by memmove() */
+    *(uct_ib_mlx5_send_wqe_bb_t *)dst = *(uct_ib_mlx5_send_wqe_bb_t *)src;
```

Comment:
1. can we use 
`UCS_WORD_COPY(uct_ib_mlx5_send_wqe_bb_t, dst, uct_ib_mlx5_send_wqe_bb_t, src, MLX5_SEND_WQE_BB);`
2. does it really guarantee ? maybe use __attribute__((nooptimize)) ?

Reply:
fixed 1.

I think it's not an absolute guarantee. The actual flag to prevent replacing assignments by `mem*()` functions is `no-tree-loop-distribute-patterns`.

But due to inlinings, seems pragma/`__attribute__` are not enforced, so to get it applied we would need all top callers to have the needed attribute to be set (even existing `restrict` keywords should allow compiler to replace at least by `memcpy()` not `memmove()`, but it's not happening).

In `rdma-core` we are populating each field in `_mlx5_post_send()`, so it will not be subject to this `memmove()` optimization.

Various options:
- remove inline from `uct_ib_mlx5_bf_copy_bb()`: seems not best
- add `volatile`, but on `-O2`, code becomes a sequence of `mov`/`add`/`jne`
- replace by `memcpy()`: seems it might not even guarantee not being replaced, and we might not be able to get `xmm`-based code.